Automatically invoke flamegraph.pl and bundle it together with executable

sergv opened the pull request:
I just felt that invoking `flamegraph.pl` automatically to get pretty svg is useful.

I also opted in for bundling `flamegraph.pl` along with executable so that there's no need to install it system-wide. Do you think that is that a good idea?

bitonic commented:
Hey, please:

* remove all cosmetic changes
* remove the switch from String to Text, until there is a reason to
* explain clearly what actual changes you performed

sergv commented:
@bitonic I removed cosmetic changes, switch from String to Text and updated description. 

Regarding switch to Text, I'm not sure I understand all of the reasons to stay with Strings. Strings are pretty inefficient for large files (as described in e.g. [here](http://www.stephendiehl.com/posts/strings.html)), and `.prof` files can get quite large. Can you please elaborate?

bitonic commented:
i know of `String`s shortcomings but i don't think it's an issue. if you feel strongly about we can do it in another pr.

thanks a lot for bearing with me, i'll review this asap (been a bit busy with work & travel.)

bitonic commented:
to elaborate a bit: i want to be able to review this quickly, and if useful changes (e.g. bundling the flame graph script) are bundled with not so useful ones (cosmetic changes, switch to text) i have to spend more time reviewing while i'd like to get the useful changes in quickly.

sergv commented:
Fair enough, decoupling changes in order to make reviews easier is definitely a good thing. Take your time to review, there's no rush.

bitonic commented:
finally got the time to review this, very helpful, thanks!

sergv commented:
Great, thanks!
